Userland/test-crypto.cpp

```diff
@@ -1480,17 +1480,114 @@ static void bigint_test_number_theory()
         }
     }
     {
-        I_TEST((Number Theory | Modular Power));
-        auto exp = Crypto::NumberTheory::ModularPower(
-            Crypto::UnsignedBigInteger::from_base10("2988348162058574136915891421498819466320163312926952423791023078876139"),
-            Crypto::UnsignedBigInteger::from_base10("2351399303373464486466122544523690094744975233415544072992656881240319"),
-            10000);
-
-        if (exp == 3059) {
-            PASS;
+        struct {
+            const char* base_b10;
+            const char* exp_b10;
+            const char* mod_b10;
+            const char* expected_b10;
+        } mod_pow_tests[] = {
+            { "2988348162058574136915891421498819466320163312926952423791023078876139", "2351399303373464486466122544523690094744975233415544072992656881240319", "10000", "3059" },
+            { "24231", "12448", "14679", "4428" },
+            { "1005404", "8352654", "8161408", "2605696" },
+            { "3665005778", "3244425589", "565668506", "524766494" },
+            { "10662083169959689657", "11605678468317533000", "1896834583057209739", "1292743154593945858" },
+            { "99667739213529524852296932424683448520", "123394910770101395416306279070921784207", "238026722756504133786938677233768788719", "197165477545023317459748215952393063201" },
+            { "49368547511968178788919424448914214709244872098814465088945281575062739912239", "25201856190991298572337188495596990852134236115562183449699512394891190792064", "45950460777961491021589776911422805972195170308651734432277141467904883064645", "39917885806532796066922509794537889114718612292469285403012781055544152450051" },
+            { "48399385336454791246880286907257136254351739111892925951016159217090949616810", "5758661760571644379364752528081901787573279669668889744323710906207949658569", "32812120644405991429173950312949738783216437173380339653152625840449006970808", "7948464125034399875323770213514649646309423451213282653637296324080400293584" },
+        };
+
+        for (auto test_case : mod_pow_tests) {
+            I_TEST((Number Theory | Modular Power));
+            auto b = Crypto::UnsignedBigInteger::from_base10(test_case.base_b10);
+            auto e = Crypto::UnsignedBigInteger::from_base10(test_case.exp_b10);
+            auto m = Crypto::UnsignedBigInteger::from_base10(test_case.mod_b10);
+            auto expected = Crypto::UnsignedBigInteger::from_base10(test_case.expected_b10);
+            auto actual = Crypto::NumberTheory::ModularPower(b, e, m);
+
+            if (actual == expected) {
+                PASS;
+            } else {
+                FAIL(Wrong result);
+                printf("b: %s\ne: %s\nm: %s\nexpect: %s\nactual: %s\n",
+                    test_case.base_b10, test_case.exp_b10, test_case.mod_b10, test_case.expected_b10, actual.to_base10().characters());
+            }
+        }
+    }
+    {
+        struct {
+            const char* base10;
```

Comment:
Just as a little usability thing, the `_bigint` UDL is already defined for these cases.
It's not constexpr or anything, but it's a bit nicer to read than `blahblah::from_base10(...)`

Reply:
Awesome! I didn't notice that.